ygen/gogen.go

```diff
@@ -730,6 +750,19 @@ func (t *{{ .Receiver }}) Get{{ .Name }}() {{ .Type }} {
 	}
 	return {{ if .IsPtr -}} * {{- end -}} t.{{ .Name }}
 }
+`
+
+	// goLeafSetterTemplate defines a template for a function that, for a
+	// particular leaf, generates a setter method.
+	goLeafSetterTemplate = `
+// Set{{ .Name }} sets the value of the leaf {{ .Name }} from the {{ .Receiver }}
+// struct. 
+func (t *{{ .Receiver }}) Set{{ .Name }}(val string) {
```

Comment:
I'm not sure that you can just take a `string` as an argument, there are different types of fields that we have within the generated structs. Given that this is the case, we need to handle different inputs here. Determining the type that is needed should likely be from the input type that is in the `generatedLeafSetter` that you created. There are some other details that you'll need though, such as the name of the generated union, such that you can call `To_XXX` for these types.

Reply:
@robshakir 
Thanks for your comments. That is a good point. I will start working on it but before that what types you would suggest to support? I will let you know if I have more questions.

Reply:
I think I figured that out. We should support whatever type that we have here: 
https://github.com/openconfig/ygot/blob/master/ygot/helpers.go

Reply:
@robshakir 

May I ask you how do you generate the expected output for the testcases? It says but it is not clear for me:
This package was generated by codegen-tests
using the following YANG input files: .....

Reply:
@robshakir 

Please take a look at this code and I think it handles any type. Please let me know what you think. I also have a test case ready but I am not sure how do you generate the expected output in advance.
	```
// goLeafSetterTemplate defines a template for a function that, for a
	// particular leaf, generates a setter method.
	goLeafSetterTemplate = `
// Set{{ .Name }} sets the value of the leaf {{ .Name }} from the {{ .Receiver }}
// struct. 
func (t *{{ .Receiver }}) Set{{ .Name }}(val interface{}) {
	if t == nil || val == nil {
		return
	}
	t.{{ .Name }} = ygot.ToPtr(val).(*{{ .Type }})
}
```
`